### ISSUE TITLE
Incorrect linting for rule LT09

### ISSUE THREAD

**Issue by james-camacho-tel360** [sqlfluff/sqlfluff#5695]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting error on nested `SELECT`; linter is claiming select targets need to be on new line when they are, in fact, on new lines

### Expected Behaviour

No linting errors

### Observed Behaviour

![image](https://github.com/sqlfluff/sqlfluff/assets/158621386/b1633e4e-8988-49ef-96bc-bf7864d78f0b)

### How to reproduce

SQL code:
```
create or alter view accounting.vw_fact_unit_asset_information as
select
    uai.unitnumber,
    uai.activitycode,
    uai.description,
    uai.depreciationytd,
    uai.netbookvalue,
    uai.salvagevalue
from (
    select
        left(ai.[Co Asset Number], 8) as unitnumber,
        ai.[Activity Code] as activitycode,
        ai.description,
        ai.[Current YTD Depr (Int)] as depreciationytd,
        isnull(ai.[Acquisition Value (Int)], 0) - isnull(ai.[Current Accum Depr (Int)], 0) as netbookvalue,
        ai.[Salvage Value (Int)] as salvagevalue,
        ai.[Current Thru Date (Int)], row_number() over (partition by left(ai.[Co Asset Number], 8) order by ai.[Current Thru Date (Int)] desc) as rn
    from sage.asset_information as ai
    where
        ai.[Co Asset Number] is not null
        and len(ai.[Co Asset Number]) = 8
        and ai.[Co Asset Number] not like '%-%'
) as uai
where uai.rn = 1;
```
In command prompt or CI pipeline:
`sqlfluff lint <filename.sql>`

### Dialect

T-SQL

### Version

sqlfluff: 2.3.5
python: 3.12.1

### Configuration

```
[sqlfluff]
dialect = tsql
exclude_rules = ST06, RF05, CP02
warnings = LT05
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Your query has a line where there are two columns defined on one line, which is leading to the _correct_ linting message:

```sql
...
ai.[Current Thru Date (Int)], row_number() over (partition by left(ai.[Co Asset Number], 8) order by ai.[Current Thru Date (Int)] desc) as rn
...
```

Once you break it into two lines, or let `sqlfluff` format it, the `LT09` message goes away (there are other warnings, but they are I guess not relevant for this issue).

I'd propose to close this issue.
